lib/valid_email/email_validator.rb

```diff
@@ -20,9 +20,14 @@ def validate_each(record,attribute,value)
       require 'valid_email/ban_disposable_email_validator'
       r = BanDisposableEmailValidator.new(:attributes => attributes, message: options[:message]).validate(record)
     end
+    # Check if part of domain is in disposable list
+    if r && options[:ban_partial_disposable_email]
+      require 'valid_email/ban_partial_disposable_email_validator'
+      r = BanPartialDisposableEmailValidator.new(:attributes => attributes, message: options[:message]).validate(record)
+    end
```

Comment:
Can it be moved to `BanDisposableEmailValidator` with options `partial: true`

Reply:
Done!